scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/fgboost/FGBoostAggregator.scala

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2021 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.ppml.fgboost
+
+
+import com.intel.analytics.bigdl.dllib.optim.{ValidationMethod, ValidationResult}
+import com.intel.analytics.bigdl.dllib.tensor.Tensor
+import com.intel.analytics.bigdl.ppml.base.StorageHolder
+import com.intel.analytics.bigdl.ppml.common.{Aggregator, FLDataType, FLPhase}
+import com.intel.analytics.bigdl.ppml.fgboost.common.{RMSEObjective, TreeObjective}
+import com.intel.analytics.bigdl.ppml.generated.FGBoostServiceProto._
+import com.intel.analytics.bigdl.ppml.generated.FlBaseProto._
+import com.intel.analytics.bigdl.ppml.utils.ProtoUtils._
+import org.apache.logging.log4j.LogManager
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+class FGBoostAggregator(validationMethods: Array[ValidationMethod[Float]] = null)
+  extends Aggregator {
+
+  val logger = LogManager.getLogger(this.getClass)
+
+  var obj: TreeObjective = new RMSEObjective
+  var nLabel = 1
+
+  val serverTreeLeaves = new ArrayBuffer[Map[Int, Float]]()
+  var validationSize = -1
+  var basePrediction: Array[Float] = null
+  val bestSplit = new java.util.HashMap[String, DataSplit]()
+  var validationResult = Array[ValidationResult]()
+
+  // wrapper methods to simplify data access
+  def getLabelStorage() = aggregateTypeMap.get(FLPhase.LABEL).getTableStorage()
+  def getSplitStorage() = aggregateTypeMap.get(FLPhase.SPLIT).getSplitStorage()
+  def getTreeLeaveStorage() = aggregateTypeMap.get(FLPhase.LEAF).getLeafStorage()
+  def getBranchStorage() = aggregateTypeMap.get(FLPhase.BRANCH).getBranchStorage()
+  def getPredictStorage() = aggregateTypeMap.get(FLPhase.PREDICT).getTableStorage()
```

Comment:
aggregateTypeMap.get(FLPhase.PREDICT).getStorage()

Reply:
Use ClassTag would bring reflection I think?

How about below instead.
```
aggregateTypeMap.get(FLPhase.PREDICT).getStorage(FLPhase.TENSOR_MAP)
```

Reply:
@jason-dai just come across one issue

If use datatype and holder, we have to use different get method, e.g. `getAStorage, getBStorage` instead of `getStorage(A)` because of the explicit return type.

If use implicit type, this seems to be the same with generic and wildcard type, and would bring casting, which go against the purpose of holder.

So seems it does not make sense if we bring them together, if holder expression is an issue of future development, I think maybe we should change the whole style from holder to generic+casting